halo2_proofs/src/plonk/circuit.rs

```diff
@@ -1331,12 +1331,12 @@ pub struct Gate<F: Field> {
 }
 
 impl<F: Field> Gate<F> {
-    pub(crate) fn name(&self) -> &'static str {
-        self.name
+    pub(crate) fn name(&self) -> String {
+        self.name.clone()
```

Comment:
It looks like the name will be cloned twice when calling `(gate_index, gate.name()).into()` to `metadata::Gate`, not sure if it'd be optimized out, but I think it'd be nice to have the code look no extra clone (and more consistent since other APIs returns also reference).

```suggestion
    pub(crate) fn name(&self) -> &str {
        self.name.as_str()
```